tutorials/time-series-visualization-with-altair/tutorial.ipynb

```diff
@@ -0,0 +1,2133 @@
+{
```

Comment:
<p>Bonus points if you figure out how to calculate and plot the <a href="https://stackoverflow.com/questions/27164114/show-confidence-limits-and-prediction-limits-in-scatter-plot" target="_blank">95% confidence limits</a> for this regression. Certainly not necessary though...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/earthengine-community/pull/152/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='90'/>

Reply:
<p>I think it is beyond the scope of this tutorial, but it would be neat to do a tutorial that focuses on sampling, time series, and stats.</p><p> </p><p>This snippet might work:</p><p> </p><pre class="ql-syntax" spellcheck="false"># example data
x = np.array([4.0,2.5,3.2,5.8,7.4,4.4,8.3,8.5])
y = np.array([2.1,4.0,1.5,6.3,5.0,5.8,8.1,7.1])
&nbsp;
# fit a curve to the data using a least squares 1st order polynomial fit
z = np.polyfit(x,y,1)
p = np.poly1d(z)
fit = p(x)
&nbsp;
# get the coordinates for the fit curve
c_y = [np.min(fit),np.max(fit)]
c_x = [np.min(x),np.max(x)]
&nbsp;
# predict y values of origional data using the fit
p_y = z[0] * x + z[1]
&nbsp;
# calculate the y-error (residuals)
y_err = y -p_y
&nbsp;
# create series of new test x-values to predict for
p_x = np.arange(np.min(x),np.max(x)+1,1)
&nbsp;
# now calculate confidence intervals for new test x-series
mean_x = np.mean(x)&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; # mean of x
n = len(x)&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; # number of samples in origional fit
t = 2.31&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; # appropriate t value (where n=9, two tailed 95%)
s_err = np.sum(np.power(y_err,2))&nbsp;&nbsp; # sum of the squares of the residuals
&nbsp;
confs = t * np.sqrt((s_err/(n-2))*(1.0/n + (np.power((p_x-mean_x),2)/
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;((np.sum(np.power(x,2)))-n*(np.power(mean_x,2))))))
&nbsp;
# now predict y based on test x-values
p_y = z[0]*p_x+z[0]
&nbsp;
# get lower and upper confidence limits based on predicted y and confidence intervals
lower = p_y - abs(confs)
upper = p_y + abs(confs)
</pre><p> </p><p> </p>

Reply:
Nice! I agree it is beyond the scope of this tutorial, and a sampling/stats tutorial in the future may be worth writing.

I moved the your snipped to a new notebook and added a chart:
https://colab.sandbox.google.com/drive/1ip37YBzAdq0Dq0RfIKM9ZggAvkrN2pm4

Reply:
Thanks for moving it to a NB and for adding the chart!